### PR TITLE
Fix scim external id in user csv download.

### PR DESCRIPTION
**Pull request by fisx**
## Description

Fixes bug: The SCIM external id is missing from the csv download when SAML is also used by the team.

Internal notes: https://wearezeta.atlassian.net/browse/SQSERVICES-534

## Checklist

 - [x] Title of this PR explains the impact of the change.
 - [x] The description provides context as to why the change should occur and
       what the code contributes to that effect. This could also be a link to a
       JIRA ticket or a Github issue, if there is one.
 - [x] The CHANGELOG.md file in the *Unreleased* section has been updated to
       explain the change which will be included in the release notes.
 - [x] If a component uses a new or changed internal endpoint of another
       component, this is mentioned in the CHANGELOG.md.
- [x] If this PR creates a new endpoint, or adds a new configuration flag, the endpoint / config-flag checklist (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.

### CODE REVIEW

**Comment by fisx** [comment]
(I'm not sure if the external Id in UserSSOId contains the *contents* of the xml NameID, or whether it sometimes still contains the entire xml blob.)

**Comment by fisx** [comment]
Also, there may be other places where we have the same bug?

File: services/galley/src/Galley/API/Teams.hs [hunk 1]
```diff
@@ -498,12 +498,15 @@ getTeamMembersCSVH (zusr ::: tid ::: _) = do
...
+    -- TODO: this is complex enough to be moved next to `U.userIdentity`, maybe as `U.userScimExternalId`.
     scimExtId :: User -> Maybe Text
-    scimExtId = U.userIdentity >=> userSSOId >=> ssoIdExtId
+    scimExtId usr = U.userIdentity >=> userSSOId >=> ssoIdExtId usr $ usr
 
-    ssoIdExtId :: UserSSOId -> Maybe Text
-    ssoIdExtId (UserSSOId _ _) = Nothing
-    ssoIdExtId (UserScimExternalId extId) = pure extId
+    ssoIdExtId :: User -> UserSSOId -> Maybe Text
+    ssoIdExtId usr (UserSSOId _ extId) = case U.userManagedBy usr of
```
Comment (smatting) [inline services/galley/src/Galley/API/Teams.hs:506]: Note that extId is
> An XML blob specifying the user's ID on the identity provider's side.

**Review by smatting — APPROVED** [review]

**Comment by fisx** [comment]
test suite will be greener with #1334 merged.

**Pull request merged**